LSPS2/README.md

```diff
@@ -0,0 +1,955 @@
+Lightning, and have the cost of their inbound liquidity be deducted from their
+first received payment.
+
+### Trust Models
+
+As of this version, there are two trust models:
+
+* The LSP trusts clients to actually claim their payment once the channel is
+  opened and the payment is forwarded.
+* The client trusts LSPs to actually open a real channel with a valid funding
+  transaction and valid funding outpoint.
+
+If the client does not claim the payment, and the funding transaction confirms,
+then the LSP will have its funds locked in a channel that was not paid for.
+Even if the LSP disables all other use of the channel until the payment paying
+for the channel is claimed, the client may then refuse all mutual close attempts
+in retaliation (by disconnecting), and force the LSP to unilaterally close,
+which locks the funds until the `to_self_delay` (indicated by the client, and
+imposed on the LSP) has passed.
+The LSP can protect against this (i.e. LSP does not trust the client) by simply
+not broadcasting the funding transaction until after it gets the preimage.
+
+Similarly, the LSP may provide a random number for the transaction ID of the
+funding transaction during opening, then forwarding as "normal" into the
+invalid channel and then receiving the preimage.
+The LSP can then spuriously fail any attempt to send out the client-received
+unds into the public network.
+The client is then unable to claim its received funds, even though the LSP
+would be able to claim the funds from the incoming forward.
+Even if the client wanted to unilaterally drop the channel on-chain, as the
+funding transaction is not valid and will never appear on the blockchain,
+it will be unable to claim the funds.
+The client can protect against this (i.e. client does not trust the LSP) by
+simply not providing the preimage until it sees the funding transaction has
+been broadcast and is in its own local mempool.
+
+If neither the client nor the LSP trust each other, they will deadlock: the
+untrusting LSP withholds the funding transaction until it sees the payment
+preimage, the untrusting client withholds the preimage until it sees the
+funding transaction.
+
+In an "LSP trusts client" model, the client MAY do any one of these things:
+
+* Immediately provide the preimage before it can validate the funding
+  transaction.
+* Wait for the funding transaction to be visible on some mempool, then
+  validate the funding transaction output is correct, and then provide the
+  preimage.
+* Wait for 1 confirmation of the funding transaction, then validate
+  the funding transaction output is correct, and then provide the
+  preimage.
+* Wait for any number of confirmations of the funding transaction, e.g.
+  3 confirmations, 6 confirmations, 100 confirmations, 2016 confirmations,
+  etc., validating the funding transaction output as soon as it can
+  see the funding transaction, and then provide the preimage.
```

Comment:
>I'd say this is bad behavior from the client, and the LSP should probably have mitigations in place to ban clients that do this?

It is required that clients are allowed to do this and that LSPs not punish them for it, otherwise LSPs have too much power over the clients and may be potentially liable outside of the protocol.